server/server.go

```diff
@@ -325,42 +332,72 @@ func (s *Server) Login(ident oidc.Identity, key string) (string, error) {
 		return ru.String(), nil
 	}
 
-	usr, err := s.UserRepo.GetByRemoteIdentity(nil, user.RemoteIdentity{
-		ConnectorID: ses.ConnectorID,
-		ID:          ses.Identity.ID,
-	})
+	remoteIdentity := user.RemoteIdentity{ConnectorID: ses.ConnectorID, ID: ses.Identity.ID}
+
+	// Get the connector used to log the user in.
+	var conn connector.Connector
```

Comment:
Breaking this out into its own method, eg .`(s *Server) GetConnectorByID(id string)` will make this long method a bit easier to read.